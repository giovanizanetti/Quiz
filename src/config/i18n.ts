import i18n from 'i18next'
import LanguageDetector from 'i18next-browser-languagedetector'
import HttpBackend from 'i18next-http-backend'
import { LANGUAGE } from '../constants'

i18n
  .use(HttpBackend)
  .use(LanguageDetector)
  .init({
    fallbackLng: LANGUAGE.en,
    interpolation: {
      escapeValue: false,
    },
    backend: {
      loadPath: '/locales/{{lng}}.json',
    },
    debug: true,
  })

export default i18n
