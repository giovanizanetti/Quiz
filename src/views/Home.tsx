import LevelSelect from '../components/LevelSelect'
import CategorySelect from '../components/CategorySelect'
import { Button, Space } from 'antd'
import { AntDesignOutlined } from '@ant-design/icons'
import { useTranslation } from 'react-i18next'
import { capitalize } from '../helpers/strings'
import { useNavigate } from 'react-router-dom'

const styleCenter = {
  display: 'flex',
  justifyContent: 'center',
  width: '100%',
}

const Home = () => {
  const { t } = useTranslation()
const navigate = useNavigate()

  const buttonLabel = capitalize(t('start-type', { type: t('quiz') }))
  const handleClick = () => navigate('quiz/question/1')

  return (
    <Space
      style={{
        ...styleCenter,
        flexDirection: 'column',
      }}
    >
      <Space wrap style={styleCenter}>
        <LevelSelect />
        <CategorySelect />
      </Space>
      <Button
        onClick={handleClick}
        style={{ margin: '2rem' }}
        type="primary"
        size="large"
        icon={<AntDesignOutlined />}
      >
        {buttonLabel}
      </Button>
    </Space>
  )
}

export default Home
