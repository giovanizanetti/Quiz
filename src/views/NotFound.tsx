import { useTranslation } from 'react-i18next'
import { capitalize } from '../helpers/strings'

export const NotFound: React.FC = () => {
  const { t } = useTranslation()

  return (
    <div>
      <h1>404 - {capitalize(t('pageNotFound'))}</h1>
      <p>{t('notFoundMessage')}</p>
    </div>
  )
}