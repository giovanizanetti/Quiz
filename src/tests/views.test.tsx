import { render } from '@testing-library/react'
import Home from '../views/Home'
import { describe, expect, it, vi } from 'vitest'
import { I18nextProvider } from 'react-i18next'
import i18n from '../config/i18n'
import NotFound from '../views/NotFound'

describe('View are rendered correctelly', () => {
  it('Home view matches the snapshot', () => {
    const { asFragment } = render(
      <I18nextProvider i18n={i18n}>
        <Home  />
      </I18nextProvider>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('Not Found view matches the snapshot', () => {
    const { asFragment } = render(
      <I18nextProvider i18n={i18n}>
        <NotFound  />
      </I18nextProvider>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
