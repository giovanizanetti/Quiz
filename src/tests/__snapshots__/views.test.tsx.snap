// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home page is rendered > matches the snapshot 1`] = `<DocumentFragment />`;

exports[`V > matches the snapshot 1`] = `<DocumentFragment />`;

exports[`View are rendered correctelly > HOmatches the snapshot 1`] = `<DocumentFragment />`;

exports[`View are rendered correctelly > Home view matches the snapshot 1`] = `<DocumentFragment />`;

exports[`View are rendered correctelly > Not FOund view matches the snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      404 - PageNotFound
    </h1>
    <p>
      notFoundMessage
    </p>
  </div>
</DocumentFragment>
`;

exports[`View are rendered correctelly > Not Found view matches the snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      404 - PageNotFound
    </h1>
    <p>
      notFoundMessage
    </p>
  </div>
</DocumentFragment>
`;

exports[`View are rendered correctelly > matches the snapshot 1`] = `<DocumentFragment />`;
